### DISCUSSION TITLE
Golang: XML Marshal conflict because of Capital Struct Name

### DISCUSSION THREAD

**Discussion by anwerj** [xuri/xgen#32] [Q&A]
Please consider this XSD for example

```
<xs:schema xmlns:xs="http://www.w3.org/2001/XMLSchema">
 <xs:complexType name="baseAddressType">
   <xs:attribute name="name"   type="xs:string" use="required"/>
   <xs:attribute name="street" type="xs:string" use="required"/>
 </xs:complexType>
</xs:schema>
```
It, because of this line of code https://github.com/xuri/xgen/blob/04f2cd700cdb4895f0cb77a86f5e6e33c045b3c9/genGo.go#L180 will convert to 
```
// BaseAddressType ...
type BaseAddressType struct {
	XMLName    xml.Name `xml:"baseAddressType"`
	NameAttr   string   `xml:"name,attr"`
	StreetAttr string   `xml:"street,attr"`
}
```

Now, if anywhere we want to use the baseAddressType, `xml.Marshal` with return conflict error. I guess it is because of the fact that go structs are Capitalized.       

Can you please suggest a fix?

**Comment by xuri** [comment]
Thanks for your feedback, I have tested and it works well, could you provide any code to reproduce this issue?

**Reply by anwerj** [reply]
Sure and Thanks for prompt reply, If we extend the schema further to use the `BaseAddressType` like 
```
<xs:schema xmlns:xs="http://www.w3.org/2001/XMLSchema">
 <xs:complexType name="baseAddressType">
   <xs:attribute name="name"   type="xs:string" use="required"/>
   <xs:attribute name="street" type="xs:string" use="required"/>
 </xs:complexType>

 <xs:element name="Address">
    <xs:complexType>
        <xs:sequence>
            <xs:element name="baseAddress" type="baseAddressType" />
        </xs:sequence>
    </xs:complexType>
 </xs:element>
</xs:schema>
```
It will convert to 
```
// BaseAddressType ...
type BaseAddressType struct {
	XMLName    xml.Name `xml:"baseAddressType"`
	NameAttr   string   `xml:"name,attr"`
	StreetAttr string   `xml:"street,attr"`
}

// Address ...
type Address struct {
	BaseAddress *BaseAddressType `xml:"baseAddress"`
}
```

Now the XML Marshal will not work on Address as there is a conflict in Name, Here is the playground code. 
https://play.golang.org/p/Upah4Q7fxAI

I would have checked for First letter case insensitively before adding XMLName field, but I have no clue how that could affect further.

**Reply by anwerj** [reply]
@xuri Got any chance of look at it, I can raise a PR.
